connector/connect/client/jvm/pom.xml

```diff
@@ -52,6 +53,12 @@
       <version>${protobuf.version}</version>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+      <scope>compile</scope>
+    </dependency>
```

Comment:
I am not that familiar with pom so maybe a question:

The common pom include the guava as well without shading it (if I am not wrong on this), so why add guava to client? client depends on common will get the guava it needs?

Reply:
Great question. This is a workaround to a pre-existing SBT issue where the resolution from the imported maven build for particular packages doesn't work for an unknown reason. This affects Spark-Connect-related projects (see [here](https://github.com/apache/spark/blob/6cae6aa5156655c79eb3f20292ccec6c479c3b1b/project/SparkBuild.scala#L667-L668) and [here](https://github.com/apache/spark/blob/6cae6aa5156655c79eb3f20292ccec6c479c3b1b/project/SparkBuild.scala#L902-L904) for example).

The pom build works fine when removing the explicit guava dep (as well as the guava exclusion from the connect-common dep) but the sbt build then fails. Thus, we explicitly mention the guava version so that SBT can then parse the version to manually include it (and it's also added as a dep in maven to ensure version consistency with sbt)

Since the issue affects multiple projects, what do you think about addressing them at once in a separate bug-fix ticket?

Reply:
Can you file a follow-up for the build issues?

Reply:
Filed a ticket here - https://issues.apache.org/jira/browse/SPARK-41967